### PR TITLE
fix #372 kube-helm README adjustments for helm3

### LINKED ISSUE #372: Instructions for installing amundsen on k8 using helm not correct?
I am trying to run this:

```
helm install dsci-amundsen templates/helm/amundsen --values impl/helm/dev/values.yaml
Error: failed to download "templates/helm/amundsen" (hint: running `helm repo update` may help)
✘-1 18:01 ~/dsci-amundsen [master L|…4] $ helm search hub templates/lyft/amundsen
No results found
```

to no avail. What can I do? What am I missing? Where is the helm chart? I'm an newb at using helm.

@javamonkey79

### PR DESCRIPTION
**Pull request by jornh**
### Summary of Changes

fix #372 kube-helm README. adjustments for helm3

### CODE REVIEW

File: amundsen-kube-helm/README.md [hunk 1]
```diff
@@ -102,9 +99,14 @@ config:
...
-With this values file, you can then setup amundsen with these commands:
+With this values file, you can then install Amundsen using Helm 2 with:
+``` shell
+helm install ./templates/helm --values impl/helm/dev/values.yaml
 ```
-helm install templates/helm/amundsen --values impl/helm/dev/values.yaml
+
+For Helm 3 it's now mandatory to specify a [chart reference name](https://helm.sh/docs/intro/using_helm/#helm-install-installing-a-package) e.g. `my-amundsen`: 
+``` shell
+helm install my-amundsen ./templates/helm --values impl/helm/dev/values.yaml
```
Comment (Jwan622) [inline amundsen-kube-helm/README.md:109]: This doesn't work for me either. What is passed into the `--values` flag? My own custom values right?

Comment (Jwan622) [reply amundsen-kube-helm/README.md:109]: I got this error when I ran the command without the --values part:
```
helm install dsci-amundsen ./templates/helm
Error: found in Chart.yaml, but missing in charts/ directory: elasticsearch
vvv

Comment (jornh) [reply amundsen-kube-helm/README.md:109]: Oh yes, you're right about the missing elasticsearch. See if running this doesn't fix it:
``` shell
helm dependency update 
```

Comment (jornh) [reply amundsen-kube-helm/README.md:109]: Regarding the `--values` flag. Yes it's to override with your own values. I just went back in git history and found an old snippet https://github.com/lyft/amundsen/tree/7c58452dc4917390602846afcf542cd285a9ae4d/amundsen-kube-helm#how-do-i-use-this that used to sit before the **"With this values file, you can then setup..."** 

Maybe something like that in an updated form should be added again?

Comment (Jwan622) [reply amundsen-kube-helm/README.md:109]: insisde the amundsen-kube-helm folder: 
```
 helm dependency update
Error: validation: chart.metadata is required
```

Comment (Jwan622) [reply amundsen-kube-helm/README.md:109]: step by step, what did you do the install this in your k8 cluster?

Comment (jornh) [reply amundsen-kube-helm/README.md:109]: Whoops sorry, I was actually running my helm dependency command from within the chart folder (`./templates/helm`) so it apparently picked up the CHART parameter with `.` as default. So inside the amundsen-kube-helm folder, try:
``` shell
helm dependency update ./templates/helm
```

Comment (jornh) [reply amundsen-kube-helm/README.md:109]: So everything from the top (as much as I can, in WSL, on a PC, with my clone directory being `C:\Hub\amundsen_220tlp_only`) - with me:

``` bash
$ pwd
/mnt/c/Hub/amundsen_220tlp_only/amundsen-kube-helm
$ helm version
version.BuildInfo{Version:"v3.1.2", GitCommit:"d878d4d45863e42fd5cff6743294a11d28a9abce", GitTreeState:"clean", GoVersion:"go1.13.8"}
$ helm dependency update ./templates/helm/
Saving 1 charts
Downloading elasticsearch from repo https://kubernetes-charts.storage.googleapis.com/
Deleting outdated charts
$ helm install my-amundsen ./templates/helm/ --debug --dry-run
install.go:158: [debug] Original chart version: ""
install.go:175: [debug] CHART PATH: /mnt/c/Hub/amundsen_220tlp_only/amundsen-kube-helm

NAME: my-amundsen
LAST DEPLOYED: Tue Apr 14 00:07:21 2020
NAMESPACE: default
STATUS: pending-install
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
{}

COMPUTED VALUES:
LONG_RANDOM_STRING: 1234
affinity: {}
dnsZone: teamname.company.com
dockerhubImagePath: amundsendev
..... 
```

Comment (Caisho) [reply amundsen-kube-helm/README.md:109]: Hi @jornh 

I am getting this error when I run the above:
```
helm install my-amundsen ./templates/helm/ --debug --dry-run
install.go:158: [debug] Original chart version: ""
install.go:175: [debug] CHART PATH: /home/joel/AISG/UDP/amundsen/amundsen-kube-helm/templates/helm

Error: unable to build kubernetes objects from release manifest: [unable to recognize "": no matches for kind "Deployment" in version "apps/v1beta1", unable to recognize "": no matches for kind "StatefulSet" in version "apps/v1beta1"]
helm.go:75: [debug] [unable to recognize "": no matches for kind "Deployment" in version "apps/v1beta1", unable to recognize "": no matches for kind "StatefulSet" in version "apps/v1beta1"]
unable to build kubernetes objects from release manifest
helm.sh/helm/v3/pkg/action.(*Install).Run
        /home/circleci/helm.sh/helm/pkg/action/install.go:247
main.runInstall
        /home/circleci/helm.sh/helm/cmd/helm/install.go:223
main.newInstallCmd.func1
        /home/circleci/helm.sh/helm/cmd/helm/install.go:117
github.com/spf13/cobra.(*Command).execute
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826
github.com/spf13/cobra.(*Command).ExecuteC
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914
github.com/spf13/cobra.(*Command).Execute
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main
        /home/circleci/helm.sh/helm/cmd/helm/helm.go:74
runtime.main
        /usr/local/go/src/runtime/proc.go:203
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1357
```

Comment (jornh) [reply amundsen-kube-helm/README.md:109]: Huh! That error isn’t giving very clear guidance ... I wonder where the `”apps/v1beta1”` comes from? Something you added in your environment at some point - from the Amundsen/elastic charts - or from helm itself? What version of helm are you on? Did it fetch the elasticsearch chart without problems?

I experienced some similarly confusing, but different, error message yesterday. It turned out I had some Amundsen Kubernetes objects lingering even after doing a `helm uninstall` of Amundsen. I found `kubectl get all --all-namespaces` to be useful there.

Maybe worth trying to do everything in a different namespace than “default” if it’s polluted in some way... (or to rule that out at least) you’d have to look up the helm syntax for it. I’m in no way a helm superuser either. I just happened to get it spun up ... unfortunately with some missteps etc. and learning along the way it’s easy to miss documenting a step.

**Comment by Jwan622** [comment]
Also,  I have the pods running. How do I visit the urls?

**Comment by jornh** [comment]
> Also, I have the pods running. 

Huzzah! Did you figure out what went wrong - after this point https://github.com/lyft/amundsen/pull/373#discussion_r407751707 (that could possibly go into a troubleshooting section)

> How do I visit the urls?

Just for testing/debugging the setup I’ve been using the `kubectl port-forward` command. (That should maybe go in NOTES.txt which is displayed after install?) For a production setup you will setup a Kubernetes ingress. I think the chart has some kind of support for that on AWS - elsewhere you’d have to provide it

**Comment by Jwan622** [comment]
I think some instructinos on how to setup both the port-forward and the production ingress would be helpful. I don't even know what a K8 ingress is and how to set it up. Some notes pointing to the chart would help too.

**Comment by jornh** [comment]
Got it. Apart from what chart pointers specifically you’re lookin for? 
The README.md already has this big table which document all values in use.

One thing I’m considering is, if while we’re  at it - and at this early stage, if it’s better to get rid of the `./templates/helm` folder levels so the `Chart.yaml` will live directly in the `/Amundsen-kube-helm` folder together with the chart `README.md`  - ping: @lukelowery @javamonkey79 as I’m not sure about helm conventions, and if it will affect packaging or other things

I’ll see what I can quickly put together. Although it’s a little unfair to expect a complete Kubernetes tutorial as part of Amundsen’s documentation... at some point the message simply has to become e.g. https://lmgtfy.com/?q=kubernetes+ingress&s=d 😉

**Comment by Jwan622** [comment]
Basically just a step-by-step readme instruction of how to visit the url of the frontend and search for prod and deployment. I have the pods running in k8 now but I can't access anything and I'm not an expert in k8. So, how would I do this?

**Comment by jornh** [comment]
So, something like this could be added to NOTES.txt:

After install you can verify the pods (or services) are running and that you can connect to them one by one through `kubectl` like this:

For services first run `kubectl get services` and notice each of the NAMEs and first number under PORT(S). Then start port-forward from k8s to your machine (either just one at at time or if you want them simultaneously either in separate terminals or by `&` backgrounding)

   ``` shell
    kubectl port-forward service/amundsen-search   5001:5001
    kubectl port-forward service/amundsen-metadata 5002:5002
    kubectl port-forward service/amundsen-frontend 5000:80  # the helm chart currently has it mapped to port :80, but my laptop has that port blocked so I move it up to :5000
    
    kubectl port-forward service/amundsen-elasticsearch 9200:9200
    kubectl port-forward service/neo4j                  7474:7474
   ```

After that you can reach each of them by browsing to
* http://localhost:5000  <----------- frontend
* http://localhost:5001/apidocs/  <-- search Swagger API 
* http://localhost:5002/apidocs/  <-- metadata Swagger API
* http://localhost:9200  <------- ElasticSearch REST API
* http://localhost:7474  <------- Neo4j Web UI

**Comment by Jwan622** [comment]
That looks awesome. And without local port-forwarding, how can I access it via the browser? It has to be a kubernetes ingress?

File: amundsen-kube-helm/README.md (hunk 1 above)
Comment (Caisho) [reply amundsen-kube-helm/README.md:109]: i think it has to do with kubernetes 1.18 under the release notes:

> the following deprecated APIs can no longer be served:
> All resources under apps/v1beta1 and apps/v1beta2 - use apps/v1 instead

Comment (jornh) [reply amundsen-kube-helm/README.md:109]: @Caisho OK, well googled! Sounds reasonable.

 It sounds like a Kubernetes 1.18 compatibility issue outside the original scope of this PR. Please open a separate issue for tracking that if you think it’s something related to Amundsen.

**Comment by jornh** [comment]
@feng-tao you can merge what’s there now. It resolves #372 and the instructions now works for helm3 as well. Thanks for asking! 👍

The additional doc improvement in https://github.com/lyft/amundsen/pull/373#issuecomment-613712340 above I’ll submit in a follow up PR later today. There not helm3 specific, just additional guidance.

**Comment by feng-tao** [comment]
thanks @jornh

**Comment by jornh** [comment]
Thanks @feng-tao!

@Jwan622 as I mentioned in https://github.com/lyft/amundsen/pull/373#issuecomment-613606883 above it’s outside the scope of Amundsen to describe how you set up ingress - as it’s a generic Kubernetes thing. Requirements for it can vary quite a bit depending on the environment.

There’s already a pointer to other info here: https://github.com/lyft/amundsen/tree/master/amundsen-kube-helm#other-notes

File: amundsen-kube-helm/README.md (hunk 1 above)
Comment (jacobhjkim) [reply amundsen-kube-helm/README.md:109]: @Caisho Maybe [this](https://github.com/lyft/amundsen/pull/483 ) PR fixed your issue.

**Pull request merged**